catalog/spatial/geocoding/spatial-geocoding-feature/src/main/java/org/codice/ddf/spatial/geocoding/feature/CatalogFeatureIndexer.java

```diff
@@ -0,0 +1,174 @@
+import org.codice.ddf.spatial.geocoding.GazetteerConstants;
+import org.opengis.feature.simple.SimpleFeature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CatalogFeatureIndexer implements FeatureIndexer {
+  private static final Logger LOGGER = LoggerFactory.getLogger(CatalogFeatureIndexer.class);
+
+  private Security security = Security.getInstance();
+
+  private CatalogFramework catalogFramework;
+
+  private CatalogHelper catalogHelper;
+
+  private IndexCallback indexCallback;
+  private int extractionCount;
+  private boolean doCreate;
+
+  private static final ThreadLocal<WKTWriter> WKT_WRITER_THREAD_LOCAL =
+      ThreadLocal.withInitial(() -> new WKTWriter());
+
+  public void setSecurity(Security security) {
+    this.security = security;
+  }
+
+  public CatalogFeatureIndexer(CatalogFramework catalogFramework, CatalogHelper catalogHelper) {
+    this.catalogFramework = catalogFramework;
+    this.catalogHelper = catalogHelper;
+  }
+
+  @Override
+  public void updateIndex(
+      String resource, FeatureExtractor featureExtractor, boolean create, IndexCallback callback)
+      throws FeatureExtractionException, FeatureIndexingException {
+    if (featureExtractor == null) {
+      throw new IllegalArgumentException("featureExtractor can't be null");
+    }
+    doCreate = create;
+    indexCallback = callback;
+    extractionCount = 0;
+
+    if (create) {
+      removeExistingMetacards();
+    }
+
+    featureExtractor.pushFeaturesToExtractionCallback(resource, this::handleFeatureExtraction);
+  }
+
+  private void handleFeatureExtraction(SimpleFeature feature) throws FeatureIndexingException {
+    createOrUpdateMetacardForFeature(feature, doCreate);
+    indexCallback.indexed(++extractionCount);
+  }
+
+  private void createOrUpdateMetacardForFeature(SimpleFeature feature, boolean create)
+      throws FeatureIndexingException {
+    try {
+      security.runWithSubjectOrElevate(
+          () -> {
+            Metacard metacard = null;
+
+            if (!create) {
+              metacard = findMetacardForFeature(feature);
+            }
+            if (metacard == null) {
+              metacard = createMetacardForFeature(feature);
+              catalogFramework.create(new CreateRequestImpl(metacard));
+            } else {
+              catalogFramework.update(new UpdateRequestImpl(metacard.getId(), metacard));
+            }
+            return null;
+          });
+    } catch (SecurityServiceException | InvocationTargetException e) {
+      LOGGER.warn("Failed to index feature", e);
+      throw new FeatureIndexingException(e.getMessage());
+    }
+  }
+
+  private Metacard createMetacardForFeature(SimpleFeature feature) throws FeatureIndexingException {
+    Metacard metacard = new MetacardImpl();
+
+    String countryCode = (String) feature.getAttribute("ISO_A3");
+    metacard.setAttribute(new AttributeImpl(Metacard.TITLE, countryCode));
```

Comment:
I think we'll want to use the new taxonomy constants for this (vs. the old ones).  So `Core.TITLE`